src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -110,6 +110,111 @@ public struct MetaData
         public int type;
     }
 
+    /// <summary>
+    /// Wrapper over all values handled by the MSIL/CLR engine.
+    /// </summary>
+    internal struct CLRStackValue
+    {
+        public bool IsEnumerable => ArrayUtils.IsEnumerable(Type);
+
+        public bool IsDouble => Value is double;
+
+        // TODO_MSIL: Figure out how to set/use this flag;
+        public bool IsDefaultArgument => false;
+
+        // TODO_MSIL: Figure out if we need to use an AddressType, like StackValue
+        public bool IsPointer => Type.IsClass;
+
+        // TODO_MSIL: Figure out how to set/use this flag;
+        public bool IsExplicitCall => false;
+
+        public bool IsNull => Value == null;
+
+        // Equivalent to StackValue.metaData.type
+        public int TypeUID { get; set; }
+
+        public System.Type Type => Value?.GetType();
+
+        // TODO_MSIL: Figure out how bad boxing/unboxing is for performance
+        public object Value { get; set; }
```

Comment:
boxing/unboxing will still be an issue.
Marshaling/Unmarshaling is the place where most of the boxing/unboxing will appen